File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/AccordionTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class AccordionTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		view.setPivotX(position < 0 ? 0 : view.getWidth());
		view.setScaleX(position < 0 ? 1F + position : 1F - position);
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/AccordionTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class AccordionTransformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        page.pivotX = (if (position < 0) 0 else page.width).toFloat()
        page.scaleX = if (position < 0) 1.toFloat() + position else 1.toFloat() - position
    }

}
